### DISCUSSION TITLE
Connecting temperature to a function

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#25887] [Q&A General] [answered]
Hello, I am trying to pass the temperature to the expression of my function to be dependant on the temperature and change value from 1 to 0 when T is less than 1100 degrees. For the temperature evolution I am using a heat source in the "Materials" block that moves over my model and it is called "vol_heat". I am not sure how to use the 'T' for my expression because it tell me that "no variable 'vol_heat' found for substitution expression". How can I pass the temperature to my expression? Thank you in advance.

  [./val_function]
    type = ParsedFunction
    expression = 'if(T<1100,1,0)'
    symbol_names = 'T'
    symbol_values = vol_heat
  [../]

**Comment by GiudGiud** [comment, marked as answer]
Hello

Functions are meant to depend on spatial location and time not really variables. 

You should look at using a material property instead. For example the ParsedMaterial 

Guillaume

**Reply by fervaldies** [reply]
Thank you! And then if I want to pass that ParsedMaterial to an AuxKernels?
```
[Materials]
  [./val_function]
    type = ParsedMaterial
    expression = 'if(T<1100,1,0)'
    coupled_variables = 'T'
  [../]
[]

[AuxKernels]
  [./val1]
    type = FunctionAux
    variable = val1
    function = val_function
  [../]
[]
```

**Reply by GiudGiud** [reply]
There is a MaterialRealAux if you want to output to an auxvariable. 
https://mooseframework.inl.gov/source/auxkernels/MaterialRealAux.html

**Reply by fervaldies** [reply]
Thanks!
